src/commands/defaults.rs

```diff
@@ -143,11 +144,22 @@ fn focus_down() {
 }
 
 fn launch_terminal() {
-    let term = env::var("WAYLAND_TERMINAL")
-        .unwrap_or("weston-terminal".to_string());
+    warn!("Got {:?}", registry::get_data("terminal"));
+    let command: String;
+    if let Ok((flags, data)) = registry::get_data("terminal").map(|d| d.resolve()) {
```

Comment:
This could be simplified, I suggest something like the following:

```
let command = registry::get_data("terminal").map(|d| d.resolve()).and_then(|(_, data)| {
        data.as_string().map(|str_ref| str_ref.to_string())
            .ok_or(RegistryError::KeyNotFound)
    }).unwrap_or("weston_terminal".to_string());
```